### PR TITLE
merge URLs into an array

### PR DESCRIPTION
**Pull request by hal-shu-sato**
## Description

- merge `downloadURL` and `downloadMirrorURL` into `downloadURLs` as an array.

## Related Issue

- #144 

## Checklist:

- [ ] ~~Updated the modification date in `mod.xml`.~~

### CODE REVIEW

File: v3/types/packages.d.ts [hunk 1]
```diff
@@ -49,13 +49,9 @@ export interface Packages {
...
     /**
-     * The URL of the package download page
+     * The URLs of the package download page
      */
-    downloadURL: string;
-    /**
-     * The URL of the mirror of the package download page
-     */
-    downloadMirrorURL?: string;
+    downloadURLs?: string[];
```
Comment (mitosagi) [inline v3/types/packages.d.ts:54]: After changing the schema, use `yarn fix` again to generate it automatically.

File: v3/schema/packages.json [hunk 2]
```diff
@@ -63,15 +63,13 @@
...
-          "downloadMirrorURL": {
-            "type": "string",
-            "pattern": "^https?:\\/\\/",
-            "description": "The URL of the mirror of the package download page"
+          "downloadURLs": {
+            "type": "array",
+            "items": {
+              "type": "string",
+              "pattern": "^https?:\\/\\/"
+            },
```
Comment (mitosagi) [inline v3/schema/packages.json:71]: It is better to specify `"minItems": 1` .

**Review by mitosagi — CHANGES_REQUESTED** [review]

**Comment by mitosagi** [comment]
Did you change `v3/data/packages.json` manually? I implemented the conversion to https://github.com/hal-shu-sato/apm/pull/469 for review, but it was difficult to compare because of the different key positions in the Json.

There are some disadvantages, such as the `id` key being moved from the beginning of the object to the middle, but how about introducing [prettier-plugin-sort-json](https://github.com/Gudahtt/prettier-plugin-sort-json) to unify the order?

**Comment by hal-shu-sato** [comment]
I fixed them.

> Did you change `v3/data/packages.json` manually? I implemented the conversion to [hal-shu-sato/apm#469](https://github.com/hal-shu-sato/apm/pull/469) for review, but it was difficult to compare because of the different key positions in the Json.

I used these code:

```js
import fs from 'fs-extra';

/**
 * adds downloadURLs and remove downloadURL and downloadMirrorURL
 */
async function run() {
  const path = 'C:\\Development\\apm-data\\v3\\data\\packages.json';
  const object = await fs.readJson(path);

  object.packages.forEach((element) => {
    const newURLs = [];
    newURLs.push(element.downloadURL);
    if ('downloadMirrorURL' in element) {
      newURLs.push(element.downloadMirrorURL);
      delete element.downloadMirrorURL;
    }
    element.downloadURL = newURLs;
    console.log(element.downloadURL);
  });

  const result = JSON.stringify(object).replaceAll(
    'downloadURL',
    'downloadURLs'
  );

  fs.writeFile(path, result);
}

run();
```

> There are some disadvantages, such as the `id` key being moved from the beginning of the object to the middle, but how about introducing [prettier-plugin-sort-json](https://github.com/Gudahtt/prettier-plugin-sort-json) to unify the order?

It is a good idea!
Let's do that.

**Review by mitosagi — APPROVED** [review]
Looks good.

I just noticed that you maintained the sort order to reduce diffs.

**Pull request merged**